### PR TITLE
Mention end2end tests in dev guide

### PR DESCRIPTION
**Pull request by jaegeral**
Documenting:
- how to run end2end tests
- how to write end2end tests and run locally
- some minor improvements on the devGuide overall (formatting)

### CODE REVIEW

File: docs/Developers-Guide.md [hunk 1]
```diff
@@ -43,32 +50,43 @@ and linting python/frontend code in your local environment you need respectively
...
 ```
+
 To run TSLint in watch mode, use
-```
+
+```python
 ! yarn run lint --watch
 ```
 
 To run a single test (there are multiple ways to do it), open a shell in the docker container:
```
Comment (kiddinn) [inline docs/Developers-Guide.md:70]: multiple ways to do it, yet only one that is provided....

what about just 

```shell
$ sudo docker exec -it $CONTAINER_ID nosetests /usr/local/src/timesketch/timesketch/lib/emojis_test.py -v
```

```

File: docs/Developers-Guide.md [hunk 2]
```diff
@@ -77,16 +95,59 @@ The unittests for the api client can use ```mock``` to emulate responses from th
 
 To introduce a new API endpoint to be tested, the endpoint needs to be registered in the ```url_router``` section in ```/api_client/python/timesketch_api_client/test_lib.py``` and the response needs to be defined in the same file.
 
+#### end2end tests
+
+End2end (e2e) tests are used at Github with every commit. Those tests will setup and run with a real elasticsearch and data imported.
```
Comment (kiddinn) [inline docs/Developers-Guide.md:100]:
```suggestion
End2end (e2e) tests are run on Github with every commit. Those tests will setup and run a full Timesketch instance, with the ability to import data and perform actions with it.
```

File: docs/Developers-Guide.md [hunk 3]
```diff
@@ -77,16 +95,59 @@ The unittests for the api client can use ```mock``` to emulate responses from th
...
+
+And the sample data (currently a plaso file and a csv) is stored in:
+
+```shell
+end_to_end_tests/test_data/
+```
+
+#### Writing end2end tests
+
+While writing end2end tests it is recommended to create a symling to the source files.
```
Comment (kiddinn) [inline docs/Developers-Guide.md:121]: `symling to ` ? are you referring to a simlink, and why simlink instead of copy to the folder

Comment (kiddinn) [reply docs/Developers-Guide.md:121]: ah... I read below... 

I would rather say `recommended` you can say `one approach to make it easier to develop these e2e tests is to create a simlink to the source files, in order for the changes to the files being reflected in the container`

Another way would be to simply mount the TS source code to this folder, the /usr/local/lib/pytho.....

File: docs/Developers-Guide.md [hunk 4]
```diff
@@ -77,16 +95,59 @@ The unittests for the api client can use ```mock``` to emulate responses from th
...
+$ export CONTAINER_ID="$(sudo -E docker container list -f name=e2e_timesketch -q)"
+$ docker exec -it $CONTAINER_ID /bin/bash
+! rm /usr/local/lib/python3.8/dist-packages/end_to_end_tests/client_test.py
+! ln -s /usr/local/src/timesketch/end_to_end_tests/client_test.py /usr/local/lib/python3.8/dist-packages/end_to_end_tests/client_test.py
+```
+
+From now on you can edit the ```client_test.py``` file outside of the docker instance and run it again with
+
+```shell
+! python3 /usr/local/src/timesketch/end_to_end_tests/tools/run_in_container.py
```
Comment (kiddinn) [inline docs/Developers-Guide.md:134]: or run with 

```shell
$ sudo docker exec -it .....
```

**Review by kiddinn — COMMENTED** [review]
few minor comments

File: docs/Developers-Guide.md (hunk 1 above)
Comment (jaegeral) [reply docs/Developers-Guide.md:70]: yes this line I have not touched to the previous just wanted to give newlines before and after the `

Comment (jaegeral) [reply docs/Developers-Guide.md:70]: but I will add it

File: docs/Developers-Guide.md (hunk 3 above)
Comment (jaegeral) [reply docs/Developers-Guide.md:121]: Good point, will add that

**Review by kiddinn — APPROVED** [review]
LGTM

**Pull request merged**